drivers/serial/uart_nrfx_uarte.c

```diff
@@ -1189,6 +1189,21 @@ static void uarte_nrfx_set_power_state(struct device *dev, u32_t new_state)
 		assert(new_state == DEVICE_PM_LOW_POWER_STATE ||
 		       new_state == DEVICE_PM_SUSPEND_STATE ||
 		       new_state == DEVICE_PM_OFF_STATE);
+
```

Comment:
I am using modem driver and using this patch does not remove all current consumption. (still 820uA remaining).
I need to add "uarte_nrfx_irq_rx_disable(dev);" here to completety remove current consumption. 
I'm not an expert in this driver so maybe somebody can help how to have a better implementation to support all cases

To have a complete working application, I also need to add "uarte_nrfx_irq_rx_enable(dev);" before line 1187

Reply:
Yep, same for us. But this isn't really the responsibility of the uart driver, it should be done in the modem driver, possibly with some help from `modem_receiver.c`. We have a patch that adds power management to the U-Blox SARA modem which does this. Alas it can't be shared (just yet anyway) due to legal reasons. The good news is that it's a pretty straightforward patch.